SConstruct

```diff
@@ -376,23 +384,55 @@ env_base.Prepend(CPPPATH=["#"])
 env_base.platform_exporters = platform_exporters
 env_base.platform_apis = platform_apis
 
-# Build type defines - more platform-specific ones can be in detect.py.
-if env_base["target"] == "release_debug" or env_base["target"] == "debug":
+# Configuration of build targets:
+# - Editor or template
+# - Debug features (DEBUG_ENABLED code)
+# - Dev only code (DEV_ENABLED code)
+# - Optimization level
+# - Debug symbols for crash traces / debuggers
+
+target = env_base["target"]
+env_base.editor_build = target.startswith("editor")
+env_base.dev_build = False
```

Comment:
For consistency, maybe this?

```suggestion
env_base.dev_build = target.endswith("dev")
```

Reply:
That's what I did initially but then I changed my mind, I don't like relying on arbitrary substrings to much unless we have a very clear pattern that both Godot and derivative projects like godot-cpp would follow.

I think that's another argument for making the `dev_build` aspect a separate SCons option.